commit 65f4cc3a6ce1472ee4092c4bbf4b19beb0a8217b
Author: Salvatore Bonaccorso <carnil@debian.org>
Date:   Fri Jan 3 12:28:47 2025 -0500

    systemd/nfs.systemd.man: Fix small wording typos
    
    The wording about what is required when configuration changes are done
    contained small wording typos. Change the sentence to "When
    configuration changes are made, ...".
    
    Fixes: 1b5881d5b9f3 ("Add nfs.systemd man page")
    Signed-off-by: Salvatore Bonaccorso <carnil@debian.org>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/systemd/nfs.systemd.man b/systemd/nfs.systemd.man
index df89ddd1..f49e7776 100644
--- a/systemd/nfs.systemd.man
+++ b/systemd/nfs.systemd.man
@@ -91,7 +91,7 @@ Most NFS daemons can be restarted at any time.  They will reload any
 state that they need, and continue servicing requests.  This is rarely
 necessary though.
 .PP
-When configuration changesare make, it can be hard to know exactly
+When configuration changes are made, it can be hard to know exactly
 which services need to be restarted to ensure that the configuration
 takes effect.  The simplest approach, which is often the best, is to
 restart everything.  To help with this, the

commit 307a75f56b565fcb07cbe26d31178cb03ee62b80
Author: Scott Mayhew <smayhew@redhat.com>
Date:   Fri Jan 3 12:26:39 2025 -0500

    conffile: add 'arg' argument to conf_remove_now()
    
    Commit 9350a97a added an optional 'arg' to section names, but the logic
    to remove configurations wasn't updated to check the 'arg' argument.
    This wasn't really a problem until commit 15e17993 updated
    conf_parse_line() to call conf_set() with override=1, the end result
    being that we'll only remember the last value seen for any given
    section/tag combination.
    
    Fixes: 9350a97a ("Added an conditional argument to the Section names")
    Fixes: 15e17993 ("conffile: process config.d directory config files.")
    Signed-off-by: Scott Mayhew <smayhew@redhat.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/support/nfs/conffile.c b/support/nfs/conffile.c
index 1e9c22b5..137fac8d 100644
--- a/support/nfs/conffile.c
+++ b/support/nfs/conffile.c
@@ -169,13 +169,15 @@ static void free_conftrans(struct conf_trans *ct)
  * Insert a tag-value combination from LINE (the equal sign is at POS)
  */
 static int
-conf_remove_now(const char *section, const char *tag)
+conf_remove_now(const char *section, const char *arg, const char *tag)
 {
 	struct conf_binding *cb, *next;
 
 	cb = LIST_FIRST(&conf_bindings[conf_hash (section)]);
 	for (; cb; cb = next) {
 		next = LIST_NEXT(cb, link);
+		if (arg && (cb->arg == NULL || strcasecmp(arg, cb->arg) != 0))
+			continue;
 		if (strcasecmp(cb->section, section) == 0
 				&& strcasecmp(cb->tag, tag) == 0) {
 			LIST_REMOVE(cb, link);
@@ -217,7 +219,7 @@ conf_set_now(const char *section, const char *arg, const char *tag,
 	struct conf_binding *node = 0;
 
 	if (override)
-		conf_remove_now(section, tag);
+		conf_remove_now(section, arg, tag);
 	else if (conf_get_section(section, arg, tag)) {
 		if (!is_default) {
 			xlog(LOG_INFO, "conf_set: duplicate tag [%s]:%s, ignoring...",
@@ -1254,7 +1256,7 @@ conf_end(int transaction, int commit)
 						node->is_default);
 					break;
 				case CONF_REMOVE:
-					conf_remove_now(node->section, node->tag);
+					conf_remove_now(node->section, node->arg, node->tag);
 					break;
 				case CONF_REMOVE_SECTION:
 					conf_remove_section_now(node->section);

commit 2ba39567ebb2a6394645ba9392a161f7a6164dd7
Author: Yongcheng Yang <yongcheng.yang@gmail.com>
Date:   Fri Jan 3 12:23:21 2025 -0500

    nfsdctl: manpage fix a few typos
    
    Reviewed-by: Benjamin Coddington <bcodding@redhat.com>
    Signed-off-by: Yongcheng Yang <yongcheng.yang@gmail.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/utils/nfsdctl/nfsdctl.8 b/utils/nfsdctl/nfsdctl.8
index 38dd1d30..b08fe803 100644
--- a/utils/nfsdctl/nfsdctl.8
+++ b/utils/nfsdctl/nfsdctl.8
@@ -2,12 +2,12 @@
 .\"     Title: nfsdctl
 .\"    Author: Jeff Layton
 .\" Generator: Asciidoctor 2.0.20
-.\"      Date: 2024-11-18
+.\"      Date: 2024-12-30
 .\"    Manual: \ \&
 .\"    Source: \ \&
 .\"  Language: English
 .\"
-.TH "NFSDCTL" "8" "2024-11-18" "\ \&" "\ \&"
+.TH "NFSDCTL" "8" "2024-12-30" "\ \&" "\ \&"
 .ie \n(.g .ds Aq \(aq
 .el       .ds Aq '
 .ss \n[.ss] 0
@@ -163,7 +163,7 @@ The fields are:
 .fam C
 + to enable a version, \- to disable
 MAJOR: the major version integer value
-MINOR: the minor version integet value
+MINOR: the minor version integer value
 .fam
 .fi
 .if n .RE
@@ -291,7 +291,7 @@ Set the pool\-mode to "pernode":
 .if n .RS 4
 .nf
 .fam C
-nfsctl pool\-mode pernode
+nfsdctl pool\-mode pernode
 .fam
 .fi
 .if n .RE
@@ -306,7 +306,7 @@ user, but configuring the server typically requires an account with elevated
 privileges.
 .SH "SEE ALSO"
 .sp
-nfs.conf(5), rpc.nfsd(8), rpc.mountd(8), exports(5), exportfs(8), nfs.conf(5), rpc.rquotad(8),  nfsstat(8), netconfig(5)
+nfs.conf(5), rpc.nfsd(8), rpc.mountd(8), exports(5), exportfs(8), rpc.rquotad(8), nfsstat(8), netconfig(5)
 .SH "AUTHOR"
 .sp
 Jeff Layton
\ No newline at end of file
diff --git a/utils/nfsdctl/nfsdctl.adoc b/utils/nfsdctl/nfsdctl.adoc
index 79f07cf0..c5921458 100644
--- a/utils/nfsdctl/nfsdctl.adoc
+++ b/utils/nfsdctl/nfsdctl.adoc
@@ -88,7 +88,7 @@ Each subcommand can also accept its own set of options and arguments. The
 
     + to enable a version, - to disable
     MAJOR: the major version integer value
-    MINOR: the minor version integet value
+    MINOR: the minor version integer value
 
   The minorversion field is optional. If not given, it will disable or enable
   all minorversions for that major version.
@@ -143,7 +143,7 @@ Change the number of running threads in first pool to 256:
 
 Set the pool-mode to "pernode":
 
-  nfsctl pool-mode pernode
+  nfsdctl pool-mode pernode
 
 == NOTES
 
@@ -157,4 +157,4 @@ privileges.
 
 == SEE ALSO
 
-nfs.conf(5), rpc.nfsd(8), rpc.mountd(8), exports(5), exportfs(8), nfs.conf(5), rpc.rquotad(8),  nfsstat(8), netconfig(5)
+nfs.conf(5), rpc.nfsd(8), rpc.mountd(8), exports(5), exportfs(8), rpc.rquotad(8), nfsstat(8), netconfig(5)

commit 5f5c0412480f48125d09cfb6a915c5f4a62b49fe
Author: Khem Raj <raj.khem@gmail.com>
Date:   Fri Jan 3 12:19:43 2025 -0500

    Fix typecast warning with clang
    
    Fixes
    file.c:200:8: error: assigning to 'char *' from 'const char *' discards qualifiers [-Werror,-Wincompatible-pointer-types-discards-qualifiers]
    
    Signed-off-by: Khem Raj <raj.khem@gmail.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/support/nsm/file.c b/support/nsm/file.c
index 27332108..5ec801c3 100644
--- a/support/nsm/file.c
+++ b/support/nsm/file.c
@@ -184,7 +184,8 @@ static char *
 nsm_make_temp_pathname(const char *pathname)
 {
 	size_t size;
-	char *path, *base;
+	char *path;
+	const char *base;
 	int len;
 
 	size = strlen(pathname) + sizeof(".new") + 1;

commit 772f7d8641f162614bba7533fedda4d3b7fc65b4
Author: Wang Mingyu <wangmy@cn.fujitsu.com>
Date:   Fri Jan 3 12:13:00 2025 -0500

    Fix const qualifier error
    
    file.c: In function nsm_make_temp_pathname:
    file.c:200:22: warning: assignment discards const qualifier from pointer target type [-Wdiscarded-qualifiers]
      200 |                 base = pathname;
          |                      ^
    
    Signed-off-by: Wang Mingyu <wangmy@cn.fujitsu.com>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/support/nsm/file.c b/support/nsm/file.c
index de122b0f..27332108 100644
--- a/support/nsm/file.c
+++ b/support/nsm/file.c
@@ -197,7 +197,7 @@ nsm_make_temp_pathname(const char *pathname)
 
 	base = strrchr(pathname, '/');
 	if (base == NULL)
-		base = pathname;
+		base = (char*)(&pathname);
 	else
 		base++;
 
